Fix the following error:

    error[E0308]: mismatched types
       --> crates/libcontainer/src/syscall/linux.rs:167:44
        |
    167 |         let res = unsafe { libc::setrlimit(rlimit.typ() as u32, rlim) };
        |                                            ^^^^^^^^^^^^^^^^^^^ expected `i32`, found `u32`
        |

--- a/crates/libcontainer/src/syscall/linux.rs
+++ b/crates/libcontainer/src/syscall/linux.rs
@@ -164,7 +164,7 @@
             rlim_cur: rlimit.soft(),
             rlim_max: rlimit.hard(),
         };
-        let res = unsafe { libc::setrlimit(rlimit.typ() as u32, rlim) };
+        let res = unsafe { libc::setrlimit(rlimit.typ() as i32, rlim) };
         if let Err(e) = Errno::result(res).map(drop) {
             bail!("Failed to set {:?}. {:?}", rlimit.typ(), e)
         }
